numba/core/bytecode.py

```diff
@@ -321,45 +361,199 @@ def get_used_globals(self):
+        if func_id.is_generator:
+            # Get the exception handler with the highest offset.
+            max_exception_target = max([e.target for e in entries])
+            # Remove any exception table entries that point to that exception
+            # handler.
+            entries = [e for e in entries if e.target != max_exception_target]
+            #breakpoint()
+
+        self.exception_entries = tuple(entries)
+
+    @property
+    def ordered_offsets(self):
+        if not self._ordered_offsets:
+            # Get an ordered list of offsets.
+            self._ordered_offsets = [o for o in self.table]
+        return self._ordered_offsets
+
+    def remove_build_list_swap_pattern(self, entries):
+        """ Find the following bytecode pattern:
+
+            BUILD_{LIST, MAP, SET}
+            SWAP(2)
+            FOR_ITER
+            ...
+            END_FOR
+            SWAP(2)
+
+            This pattern indicates that a list/dict/set comprehension has
+            been inlined. In this case we can skip the exception blocks
+            entirely along with the dead exceptions that it points to.
+            A pair of exception that sandwiches these exception will
+            also be clubbed into a single exception.
+        """
+
+        stack = []
+
+        def pop_and_club_exceptions(entries: list, end_entry):
+            start_entry = stack.pop()
+            start_offset = start_entry.start
+            target_offset = start_entry.target
+            end_offset = end_entry.end
+
+            to_be_removed_entries = [
+                e for e in entries if (e.start >= start_offset and
+                                       e.end <= end_offset and
+                                       e.target == target_offset)
+            ]
+
+            lower_entry_idx = entries.index(to_be_removed_entries[0]) - 1
+            upper_entry_idx = entries.index(to_be_removed_entries[-1]) + 1
+
+            if lower_entry_idx >= 0 and upper_entry_idx < len(entries):
+                lower_entry = entries[lower_entry_idx]
+                upper_entry = entries[upper_entry_idx]
+                if lower_entry.target == upper_entry.target:
+                    entries[lower_entry_idx] = _ExceptionTableEntry(
+                        lower_entry.start,
+                        upper_entry.end,
+                        lower_entry.target,
+                        lower_entry.depth,
+                        upper_entry.lasti)
+                    entries.remove(upper_entry)
+
+            dead_entries = [
+                e for e in entries if e.start == target_offset
+            ]
+
+            entries = [e for e in entries if e not in to_be_removed_entries]
+            entries = [e for e in entries if e not in dead_entries]
+            return entries
+
+        for offset, inst in self.table.items():
+            # Check start of pattern, three instructions.
+            # Work out the index of the insteruction.
```

Comment:
```suggestion
            # Work out the index of the instruction.
```

Reply:
Fixed in 0836b63048646a6d3d4ac91276468ea989d201a4